internal/controller/kbsconfig_controller.go

```diff
@@ -635,26 +637,118 @@ func (r *KbsConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		r.namespace = KbsOperatorNamespace
 	}
 
+	configMapMapper, err := configMapToKbsConfigMapper(r.Client, r.log)
+	if err != nil {
+		return err
+	}
+
+	secretMapper, err := secretToKbsConfigMapper(r.Client, r.log)
+	if err != nil {
+		return err
+	}
+
 	// Create a new controller and add a watch for KbsConfig including the following secondary resources:
 	// KbsConfigMap, KbsSecret, KbsAsConfigMap, KbsRvpsConfigMap in the same namespace as the controller
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&confidentialcontainersorgv1alpha1.KbsConfig{}).
 		// Watch for changes to ConfigMap, Secret that are in the same namespace as the controller
 		// The ConfigMap and Secret are not owned by the KbsConfig
 		Watches(
-			&source.Kind{Type: &corev1.ConfigMap{}},
-			&handler.EnqueueRequestForObject{},
+			&corev1.ConfigMap{},
+			handler.EnqueueRequestsFromMapFunc(configMapMapper),
 			builder.WithPredicates(namespacePredicate(r.namespace)),
 		).
 		Watches(
-			&source.Kind{Type: &corev1.Secret{}},
-			&handler.EnqueueRequestForObject{},
+			&corev1.Secret{},
+			handler.EnqueueRequestsFromMapFunc(secretMapper),
 			builder.WithPredicates(namespacePredicate(r.namespace)),
 		).
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Secret{}).
 		Complete(r)
+}
+
+// create mapper to transform from ConfigMap to KbsConfig
+func configMapToKbsConfigMapper(c client.Client, log logr.Logger) (handler.MapFunc, error) {
+	mapperFunc := func(ctx context.Context, o client.Object) []reconcile.Request {
+		log.Info("Got a configMap object")
+		configMap, ok := o.(*corev1.ConfigMap)
+		if !ok {
+			log.Info("Object is not a ConfigMap")
+			return nil
+		}
+
+		// Get the KbsConfig object
+		kbsConfigList := &confidentialcontainersorgv1alpha1.KbsConfigList{}
+		err := c.List(ctx, kbsConfigList, client.InNamespace(configMap.Namespace))
+		if err != nil {
+			log.Info("Error in listing KbsConfig", "err", err)
+			return nil
+		}
+
+		log.Info("Checking KbsConfig", "ConfigMap.Name", configMap.Name, "KbsConfigList", kbsConfigList.Items)
+
+		var requests []reconcile.Request
+		for _, kbsConfig := range kbsConfigList.Items {
+			if kbsConfig.Spec.KbsConfigMapName == configMap.Name ||
```

Comment:
Maybe using a stringset is more readable?
For example:
s := stringset.New("lead", "iron", "copper", "chromium")
stringset.Contains(s, "chromium")

Reply:
Nice. I'll change it

Reply:
Had second thoughts while implementing it. First stringset doesn't seem to be a native package. Need to rely on third party libs which seems to be unmaintained.
Also since we need to initialise the stringset for every iteration in the loop, we are not gaining anything. Probably best suited for constants.